aea/protocols/dialogue/base.py

```diff
@@ -132,11 +131,7 @@ def dialogue_starter_addr(self) -> str:
     def __eq__(self, other: Any) -> bool:
         """Check for equality between two DialogueLabel objects."""
         if isinstance(other, DialogueLabel):
-            return (
-                self.dialogue_reference == other.dialogue_reference
-                and self.dialogue_starter_addr == other.dialogue_starter_addr
-                and self.dialogue_opponent_addr == other.dialogue_opponent_addr
-            )
+            return hash(self) == hash(other)
```

Comment:
Explain?

Reply:
dont remember right now but assume sloted objects will have unique hash over every slot field

Reply:
anyway need to check, cause implemented long time ago

Reply:
did you check now?